gax/backoff_policy.h

```diff
@@ -0,0 +1,164 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_GAX_BACKOFF_POLICY_H_
+#define GOOGLE_GAX_BACKOFF_POLICY_H_
+
+#include <chrono>
+#include <memory>
+#include <random>
+
+// Note: gtest-1.8.1 doesn't have a separate target for gtest_prod.
+//       Rather than add an overlarge dependency on @gtest//:gtest,
+//       just define the necessary macro ourselves.
+#define FRIEND_TEST(test_case_name, test_name)\
```

Comment:
What about putting this definition into its own header file under gax/internal? Then we will be able to include it from other files as well.

Reply:
Good call. Redefining the macro ourselves makes for a smoother transition if we update our gtest dependency, but there's no reason to make additional work.